core/execution/future/market.go

```diff
@@ -1652,7 +1652,7 @@ func (m *Market) leaveAuction(ctx context.Context, now time.Time) {
 		}
```

Comment:
We would have already triggered monitoring auction above, is that ok or should it be handled too as part of this fix?

Reply:
If by that you mean that we've already entered monitoring auction by the time we reach this point, then yes, we definitely want to cover that case here. It used to be a check for a nil/zero mark price when leaving opening auction, and we assumed we would not be in auction if that happened. With the composite mark price stuff, that's no longer a guarantee and therefore the check allows for the auction state to be in auction, albeit not an opening auction.

Reply:
the problem @EVODelavega potentially is that we've entered monitoring auction (in the markPriceCalculator), and here we'd call enterAuction again.